### PR TITLE
fix: add title to Vectara upload

### PR DESCRIPTION
**Pull request by potter-potter**
Small improvement to Vectara requested by Ofer at Vectara

In the "Document" construct, every document can have a title. If it's there, in the UI it will show up above the document (otherwise you get "Untitled")

### CODE REVIEW

File: unstructured/ingest/connector/vectara.py [hunk 1]
```diff
@@ -230,9 +230,12 @@ def get_metadata(element) -> t.Dict[str, t.Any]:
             local_path = doc._output_filename
             with open(local_path) as json_file:
                 dict_content = json.load(json_file)
-                [e["type"] for e in dict_content]
                 vdoc = {
                     "documentId": str(uuid.uuid4()),
+                    "title": dict_content[0]
+                    .get("metadata", {})
+                    .get("data_source", {})
+                    .get("url", "Untitled"),
```
[the remark this answers is not in the document]
Comment (potter-potter) [reply unstructured/ingest/connector/vectara.py:238]: Good question. Currently everything is Untitled as per my understanding. And thus we get no title in the results. I believe he suggested the URL which would be unique if 2 documents were named the same when you uploaded. Do you think we should default to "filename" instead? Or should we look for url and then look for filename?

When we add title we get (currently the title is not filled in):
Screenshot 2024-02-05 at 12 03 07 PM

[the remark this answers is not in the document]
Comment (ryannikolaidis) [reply unstructured/ingest/connector/vectara.py:238]: I guess it was just strange for a url to be "Untitled".  can this be a null value if not found? or maybe just empty string if not?

> Currently everything is Untitled as per my understanding. 

Meaning urls in the Unstructured result are coming through as "Untitled"? Or is this something specific to defaults in Vectara?

[the remark this answers is not in the document]
Comment (potter-potter) [reply unstructured/ingest/connector/vectara.py:238]: Got it. Yeah totally. It should just be None. It _is_ kind of strange to declare something as Untitled.

[the remark this answers is not in the document]
Comment (potter-potter) [reply unstructured/ingest/connector/vectara.py]: Confirmed with Ofer. Should be None. Also tested. Works as before this PR with None.

**Review by ryannikolaidis — APPROVED** [review]
lgtm!

File: unstructured/ingest/connector/vectara.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryannikolaidis) [reply unstructured/ingest/connector/vectara.py]: Perfect, thanks for digging there!

**Pull request merged**